File: src/me/krickl/memebotj/InternalCommands/UserCommands/UptimeCommand.scala
package me.krickl.memebotj.InternalCommands.UserCommands

import java.text.SimpleDateFormat
import java.util.Date

import me.krickl.memebotj.{ChannelHandler, CommandHandler, UserHandler}

class UptimeCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel: String, command: String, dbprefix: String) {
  this.setNeededCommandPower(75)

  override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
    val currentUpTimeSeconds = System.currentTimeMillis - channelHandler.getStreamStartTime
    val date = new Date(currentUpTimeSeconds)
    val formattedTime = new SimpleDateFormat("HH:mm:ss").format(date)
    channelHandler.sendMessage(f"Uptime: $formattedTime", this.getChannelOrigin)
  }
}

File: src/main/java/me/krickl/memebotj/Commands/Internal/UptimeCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.Channel.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.User.UserHandler;

/**
 * This file is part of memebotj.
 * Created by unlink on 03/05/16.
 */
public class UptimeCommand extends CommandHandler {
    public UptimeCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        // fuck it let's just use this
    }
}
